app/services/databaseService.js

```diff
@@ -8,26 +8,42 @@ const sequelize = new Sequelize(
 )
 // Define the Model
 const PaymentDataModel = sequelize.define('payment_activity_data', {
-  id: { type: DataTypes.INTEGER, primaryKey: true },
   payee_name: DataTypes.STRING(32),
   part_postcode: DataTypes.STRING(8),
   town: DataTypes.STRING(32),
-  parliamentary_constituency: DataTypes.STRING(32),
   county_council: DataTypes.STRING(64),
-  scheme: DataTypes.STRING(64),
-  scheme_detail: DataTypes.STRING(128),
-  activity_level: DataTypes.STRING(16),
   amount: DataTypes.DOUBLE
 })
 
-// Collect and display the db restuls
-async function getPaymentData () {
+// Collect the DB restuls
+async function getPaymentData (searchString = '', limit = 20, offset = 0) {
+  if (searchString === '') throw new Error('Empty search content')
+  const mf = 0.4 // Matching factor
+  const whereClause = {
+    [Op.or]: [
+      where(fn('SIMILARITY', col('payee_name'), searchString), { [Op.gt]: mf }),
+      where(fn('SIMILARITY', col('part_postcode'), searchString), { [Op.gt]: mf }),
+      where(fn('SIMILARITY', col('town'), searchString), { [Op.gt]: mf }),
+      where(fn('SIMILARITY', col('county_council'), searchString), { [Op.gt]: mf })
+    ]
+  }
+
   try {
-    return PaymentDataModel.findAll()
+    const result = await PaymentDataModel.findAndCountAll({
+      limit: limit,
+      offset: offset,
+      where: whereClause,
+      group: ['payee_name', 'part_postcode', 'town', 'county_council'],
+      attributes: [
+        'payee_name', 'part_postcode', 'town', 'county_council',
+        [sequelize.fn('sum', sequelize.col('amount')), 'total-amount']
```

Comment:
Since we are using underscore in everywhere else (such as payee_name, part_postcode), I think total-count should be total_count? What do you think?

Reply:
agreed, and updated the code now.